components/script/dom/node.rs

```diff
@@ -2523,6 +2523,11 @@ pub enum ChildrenMutation<'a> {
         next: Option<&'a Node>,
     },
     ReplaceAll { removed: &'a [&'a Node], added: &'a [&'a Node] },
+    /// Mutation for when a CharacterData node's data is modified.
+    /// This doesn't change the structure of the list, which is what the other
+    /// variants' fields are stored for at the moment, so this can just have no
+    /// arguments.
+    CharacterData,
```

Comment:
This should be `Text`, we never have to reparse anything if a `Comment` or a `ProcessingInstruction` is changed.